metrics/src/main/kotlin/com/cosmotech/metrics/MetricsServiceImpl.kt

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.metrics
+
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.events.PersistentMetricEvent
+import com.cosmotech.api.metrics.PersistentMetric
+import com.redislabs.redistimeseries.RedisTimeSeries
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Service
+import redis.clients.jedis.JedisPool
+
+private const val MILLISECONDS_IN_DAY = 86400000
+private const val KEY_PREFIX = "ts"
+
+@Service
+class MetricsServiceImpl(
+    private val jedisPool: JedisPool,
+    private val timeSeries: RedisTimeSeries,
+    private val csmPlatformProperties: CsmPlatformProperties,
+) : MetricsService {
+
+  private val logger = LoggerFactory.getLogger(MetricsServiceImpl::class.java)
+
+  override fun storeMetric(metric: PersistentMetric) {
+    createTimeSeries(metric)
+    addMetricToTimeSeries(metric)
+  }
+
+  private fun addMetricToTimeSeries(metric: PersistentMetric) {
+    logger.debug("METRICS: addMetricToTimeSeries")
+    val key = getMetricKey(metric)
+    if (metric.incrementBy > 0 && metric.value > 0) {
+      throw IllegalArgumentException("Cannot set both incrementBy and value")
+    }
+
+    val timestamp =
+        when (metric.incrementBy) {
+          0 ->
+              timeSeries.add(
+                  key,
+                  metric.timestamp,
+                  metric.value,
+              )
+          else ->
+              timeSeries.incrBy(
+                  key,
+                  metric.incrementBy,
+                  metric.timestamp,
+              )
+        }
+
+    logger.debug("METRICS: addMetricToTimeSeries done for $key at $timestamp")
+  }
+
+  @Suppress("EmptyElseBlock")
+  private fun createTimeSeries(metric: PersistentMetric) {
+    logger.debug("METRICS: createTimeSeries")
```

Comment:
Can you remove the log please ?

Reply:
why remove the logs? I prefer to keep them